vaadin-button-flow-parent/vaadin-button-flow/src/main/resources/META-INF/resources/frontend/buttonFunctions.js

```diff
@@ -0,0 +1,12 @@
+window.Vaadin.Flow.button = {
+  initDisableOnClick: (button) => {
+    if (!button.__hasDisableOnClickListener) {
+      button.addEventListener('click', disableOnClickListener(button));
+      button.__hasDisableOnClickListener = true;
+    }
+  }
+}
+
+function disableOnClickListener({currentTarget: button}) {
+  button.disabled = button.hasAttribute('disableOnClick');
+}
```

Comment:
nit: it's [recommended](https://typescript-eslint.io/rules/no-use-before-define/) to define the function above from where it's used so I'd suggest moving this function above the `window.Vaadin.Flow.button`.